src/main/java/net/snowflake/ingest/streaming/internal/SnowflakeStreamingIngestChannelFactory.java

```diff
@@ -119,19 +116,7 @@ SnowflakeStreamingIngestChannelInternal<T> build() {
           this.encryptionKeyId,
           this.onErrorOption,
           this.defaultTimezone,
-          this.owningClient.getParameterProvider().getBlobFormatVersion(),
-          allocator);
-    }
-
-    private BufferAllocator createBufferAllocator() {
```

Comment:
So the memory allocation logic is inside the Parquet library? What if we want to use a different allocator per channel/table for usage tracking and better estimation?

Reply:
> So the memory allocation logic is inside the Parquet library?

Yes.

> What if we want to use a different allocator per channel/table for usage tracking and better estimation?

That's a good point, we might need it.